solutions/usaco-673.mdx

```diff
@@ -0,0 +1,92 @@
+---
+id: usaco-673
+source: USACO Plat 2016 December
+title: Team Building
+author: Ryan Chou
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_team_platinum_dec16.html)
+
+Let's say that we want to find the answer for some subsection of the sample input. Then, we would need the number of elements to consider for Farmer John and Farmer Paul, and the size of the team. A naive approach would be to go through these in $NMK$ time, and due to the low bounds on $N$, $M$, and $K$, this'll actually run in time. So, we'll just run a bottom-up DP where
+
+$\texttt{dp}[i][j][k]$ = the number of teams of size $k$ after we've considered the first $i$ cows in Farmer John's team, the first $j$ cows in Farmer Paul's team.
+
+Then, our transition would be:
+
+$\texttt{dp}[i][j][k]$ = $\texttt{dp}[i - 1][j][k] + \texttt{dp}[i][j - 1][k] - \texttt{dp}[i - 1][j - 1][k]$
```

Comment:
how does this work?

Reply:
aka what do each of the numbers mean, and why are we adding/subtracting them?

Reply:
ok, ill add a comment

Reply:
don't see it anywhere?

Reply:
huh, i thought i added it?

ohh wait...